hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminBackoffMonitor.java

```diff
@@ -665,6 +672,16 @@ private void scanDatanodeStorage(DatanodeDescriptor dn,
             blockList.put(b, null);
           }
           numBlocksChecked++;
+          if (Time.monotonicNow() - beginTime > scanDatanodeStorageLockTimeMs) {
+            namesystem.readUnlock();
+            try {
+              Thread.sleep(1);
```

Comment:
Is 1ms sleep time enough to remove lock starvation or we should add bit more?

Reply:
> Is 1ms sleep time enough to remove lock starvation or we should add bit more?

Thanks @virajjasani for your review, I am trying a new implementation now, this code may be replaced.